source/code/plugins/lib_patch_management.rb

```diff
@@ -0,0 +1,154 @@
+require 'rexml/document'
+require 'cgi'
+require 'digest'
+require 'set'
+
+require_relative 'oms_common'
+
+class LinuxUpdates
+
+    @@delimiter = "_"
+    @@hostOSDetails = `cat /etc/opt/microsoft/scx/conf/scx-release`.split(/\r?\n/)
+    @@hostOSDetailsMap = Hash[(@@hostOSDetails.map {|s| s.split("=")}).map {|key, value| [key, value]}]
+
+    @@log = nil
+    def self.log= (value)
+        @@log = value
+    end
+
+    @@force_send_last_upload = Time.now
+
+    @@prev_hash = ""
+    def self.prev_hash= (value)
+        @@prev_hash = value
+    end
+
+    def self.instanceXMLtoHash(instanceXML)
+        ret = {}
+        propertyXPath = "PROPERTY"
+        instanceXML.elements.each(propertyXPath) { |inst| #AnonFunc in rb.
+            name = inst.attributes['NAME']
+            rexmlText = REXML::XPath.first(inst, 'VALUE').get_text # TODO escape unicode chars like "&amp;"
+            value = rexmlText ? rexmlText.value.strip : ''
+            ret[name] = value
+        }
+        ret
+    end
+
+    def self.availableUpdatesXMLtoHash(availableUpdatesXML)
+        availableUpdatesHash = instanceXMLtoHash(availableUpdatesXML)
+        ret = {}
+        
+        ret["CollectionName"] = availableUpdatesHash["Name"] + @@delimiter + 
+                                availableUpdatesHash["Version"] + @@delimiter + 
+                                @@hostOSDetailsMap["OSFullName"]
+        ret["PackageName"] = availableUpdatesHash["Name"]
+        ret["PackageVersion"] = availableUpdatesHash["Version"]
+        ret["Timestamp"] = OMS::Common.format_time(availableUpdatesHash["BuildDate"].to_i)
+        ret["Repository"] = availableUpdatesHash.key?("Repository") ? availableUpdatesHash["Repository"] : ""
+        ret["Installed"] = false
+        ret
+    end
+
+    def self.installedPackageXMLtoHash(packageXML)
+        packageHash = instanceXMLtoHash(packageXML)
+        ret = {}
+        
+        ret["CollectionName"] = packageHash["Name"] + @@delimiter + 
+                                packageHash["Version"] + @@delimiter + 
+                                @@hostOSDetailsMap["OSFullName"]
+        ret["PackageName"] = packageHash["Name"]
+        ret["PackageVersion"] = packageHash["Version"]
+        ret["Timestamp"] = OMS::Common.format_time(packageHash["InstalledOn"].to_i)
+        ret["Size"] = packageHash["Size"]
+        ret["Repository"] = packageHash.key?("Repository") ? packageHash["Repository"] : "" 
+        ret["Installed"] = true
+        ret
+    end
+
+    def self.strToXML(xml_string)
+        xml_unescaped_string = CGI::unescapeHTML(xml_string)
+        REXML::Document.new xml_unescaped_string
+    end
+
+    # Returns an array of xml instances (all types)
+    def self.getInstancesXML(inventoryXML)
```

Comment:
@niroyb 
Yes. I agree. These should go into a common_utilities file which can be used by the all the subsequent plugin_libraries. I can either import the lib_changetracking or refactor both libraries and create common_utils now. Please let me know if you have a preference.